File: remote_config.m
#import "FirebaseRemoteConfig/FirebaseRemoteConfig.h"

// Get the template
FIRRemoteConfig *remoteConfig = [FIRRemoteConfig remoteConfig];
[remoteConfig getTemplateWithCompletion:^(FIRRemoteConfigTemplate *template, NSError *error) {
  if (error) {
    NSLog(@"Error getting template: %@", error);
    return;
  }

  // See the ETag of the template
  NSLog(@"ETag: %@", template.ETag);

  // Get a parameter
  FIRRemoteConfigParameter *parameter = template.parameters[@"header_text"];
  NSLog(@"%@", parameter.defaultValue.stringValue);

  // Get a condition
  FIRRemoteConfigCondition *condition = template.conditions[0];
  NSLog(@"%@", condition.name);
  NSLog(@"%@", condition.expression);
  NSLog(@"%@", @(condition.tagColor));

  // Publish the template
  [remoteConfig publishTemplate:template completion:^(FIRRemoteConfigTemplate *publishedTemplate, NSError *error) {
    if (error) {
      NSLog(@"Error publishing template: %@", error);
      return;
    }

    NSLog(@"Template has been published");
    NSLog(@"ETag: %@", publishedTemplate.ETag);
  }];
}];

File: remote_config.swift
import FirebaseRemoteConfig

// Get the template
let remoteConfig = RemoteConfig.remoteConfig()
remoteConfig.getTemplate { (template, error) in
  if let error = error {
    print("Error getting template: \(error)")
    return
  }

  // See the ETag of the template
  print("ETag: \(template!.eTag)")

  // Get a parameter
  let parameter = template!.parameters["header_text"]!
  print("\(parameter.defaultValue!.stringValue!)")

  // Get a condition
  let condition = template!.conditions[0]
  print("\(condition.name)")
  print("\(condition.expression)")
  print("\(condition.tagColor)")

  // Publish the template
  remoteConfig.publish(template!) { (publishedTemplate, error) in
    if let error = error {
      print("Error publishing template: \(error)")
      return
    }

    print("Template has been published")
    print("ETag: \(publishedTemplate!.eTag)")
  }
}
